client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
+      <template v-for="week in ['total', ...selectedWeeks]" #[`item.week_${week}`]="{item, value}">
+        <v-tooltip :key="`${item.username}_week_${week}_wins`" top v-if="value">
+          <template #activator="{on, attrs}">
+            <v-chip
+              :key="`${item.username}_week_${week}`"
+              :color="colorForScoreByWeek(week, value)"
+              dark
+              :outlined="['primary', '#000'].includes(colorForScoreByWeek(week, value))"
+              v-bind="{
+                ...attrs,
+                ...dataAttribute(item.username, week),
+              }"
+              v-on="on"
+            >
+              {{ tableCell(item, week) }}
+            </v-chip>
+          </template>
+          {{ playersBeaten(item.username, week) }}
+        </v-tooltip>
+      </template>
+    </v-data-table>
+  </div>
+</template>
+<script>
+const Result = {
+  WON: 1,
+  LOST: 2,
+  INCOMPLETE: -1,
+};
+
+export default {
+  name: 'StatsLeaderboard',
+  props: {
+    loading: Boolean,
+    season: {
+      type: Object,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      sortBy: 'rank',
+      metricChoices: ['Points and Wins', 'Points Only', 'Wins Only'],
+      selectedMetric: 'Points and Wins',
+      weeks: [
+        { text: 'Week 1', value: 1 },
+        { text: 'Week 2', value: 2 },
+        { text: 'Week 3', value: 3 },
+        { text: 'Week 4', value: 4 },
+        { text: 'Week 5', value: 5 },
+        { text: 'Week 6', value: 6 },
+        { text: 'Week 7', value: 7 },
+        { text: 'Week 8', value: 8 },
+        { text: 'Week 9', value: 9 },
+        { text: 'Week 10', value: 10 },
+        { text: 'Week 11', value: 11 },
+        { text: 'Week 12', value: 12 },
+        { text: 'Week 13', value: 13 },
+      ],
+      selectedWeeks: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
+    };
+  },
+  computed: {
+    tableColumns() {
+      if (!this.season || !this.season.rankings || this.season.rankings.length === 0) {
+        return [];
+      }
+      const res = [
+        { text: 'User', value: 'username' },
+        { text: 'Rank', value: 'rank' },
+        { text: 'Season Total', value: 'week_total' },
+      ];
+      // Add headers for each week
+      for (const weekNum of this.selectedWeeks) {
+        res.push({
+          text: `Week ${weekNum}`,
+          value: `week_${weekNum}`,
+        });
+      }
+      return res;
+    },
```

Comment:
I think organizing everything by columns, rows, etc, is a good option. I'd need to spend a lot more time looking it over. I think mainly it's probably worth reducing complexity where you can and making sure the code stays readable. You can always reduce complexity by pulling out code to separate functions and such at a later date during a refactor. Something like this:
```js
    tableColumns() {
        if (!this.season || !this.season.rankings || this.season.rankings.length === 0) {
            return [];
        }
        return [
            { text: 'User', value: 'username' },
            { text: 'Rank', value: 'rank' },
            { text: 'Season Total', value: 'week_total' },
            ...(this.selectedWeeks.map(weekNum => ({
                text: `Week ${weekNum}`,
                value: `week_${weekNum}`,
            }))),
        ];
    },
```

Reply:
I like the use of spread and map here over the for loop